include/oboe/AudioStreamBuilder.h

```diff
@@ -305,9 +305,45 @@ class AudioStreamBuilder : public AudioStreamBase {
         return this;
     }
 
+    /**
+     * Specifies an object to handle data related callbacks from the underlying API.
+     *
+     * <strong>Important: See AudioStreamCallback for restrictions on what may be called
+     * from the callback methods.</strong>
+     *
+     * @param dataCallback
+     * @return pointer to the builder so calls can be chained
+     */
+    AudioStreamBuilder *setDataCallback(oboe::AudioStreamDataCallback *dataCallback) {
```

Comment:
Random thought. For both the data and error callbacks the most common use case is that you just implement `onAudioReady` and `onErrorAfterClose`. I wonder if we should overload `setDataCallback` to take a lambda of type: `AudioCallbackResult(AudioStream *, void *, int)`. This would enable you to supply a callback function, rather than a whole class. Same for `setErrorCallback`.

Reply:
Good idea for a subsequent patch.